yrs/src/block.rs

```diff
@@ -1112,13 +1112,14 @@ impl SplittableString {
             OffsetKind::Bytes => {
                 let mut remaining = offset;
                 let mut i = 0;
-                for c in self.content.encode_utf16() {
+                // since this offset is used to splitting later on - and we can only split entire
+                // characters - we're computing by characters
+                for c in self.content.chars() {
                     if remaining == 0 {
                         break;
                     }
-                    let utf8_len = if c < 0x80 { 1 } else { 2 };
-                    remaining -= utf8_len;
-                    i += 1;
+                    remaining -= c.len_utf8() as u32;
+                    i += c.len_utf16() as u32;
```

Comment:
Here we're mapping offset provided by user in UTF-8 bytes (`OffsetKind::Bytes`), while what we need to return is a value provided in UTF-16 (since its unit of measure for Yjs item.id.clock).